relations/src/r1cs/constraint_system.rs

```diff
@@ -473,16 +504,14 @@ impl<F: Field> ConstraintSystem<F> {
         }
     }
 
-    /// Reduce the constraint weight.
-    ///
-    /// At this moment, it is a wrapper to `outline_lcs`.
-    /// More weight reductions may be added later.
-    ///
-    /// Useful for SNARKs like [\[Marlin\]](https://eprint.iacr.org/2019/1047) or
-    /// [\[Fractal\]](https://eprint.iacr.org/2019/1076), where addition gates
-    /// are not cheap.
-    pub fn reduce_constraint_weight(&mut self) {
-        self.outline_lcs();
+    /// Finalize the constraint system (either by outlining or inlining,
+    /// if an optimization goal is set).
+    pub fn finalize(&mut self) {
+        match self.optimization_goal {
+            OptimizationGoal::None => (),
```

Comment:
This line would need to be changed into `OptimizationGoal::None =>  self.inline_all_lcs(),`